### PR TITLE
Add support for multiple nested modals.

### PR DESCRIPTION
**Pull request by schryer**
These changes allow for multiple nested modals.

### CODE REVIEW

**Comment by schryer** [comment]
This addresses both the body class removal issue and the padding issue when multiple modals were used at the same time.

**Comment by schryer** [comment]
I have not squashed these manually because GitHub squashing does not work with your workflow tools.

File: src/toggle-class.js [hunk 1]
```diff
@@ -1,11 +1,27 @@
 
+// This global variable is required so that the body class is removed only
```
Comment (Jyrno42) [inline src/toggle-class.js]: Why do this here? Whats preventing you of tracking the count of opened modals in the call site?

File: src/components/Modal.js [hunk 2]
```diff
@@ -80,6 +78,12 @@ class Modal extends Component {
 
         if (this.props.isOpen && !this.props.isStatic) {
             if (this.props.onCancel) {
+
+                // Remove open class.
+                const container = document.body;
```
Comment (Jyrno42) [inline src/components/Modal.js]: Modal.js should not contain anything related to the dom. If you fix the logic inside `BrowserModal.onToggle` this change here is not required

File: src/browser.js [hunk 3]
```diff
@@ -6,6 +6,9 @@ import Modal from './components/Modal';
 import toggleClass from './toggle-class';
 import getScrollbarSize from './utils/scrollbarSize';
 
+// This keeps track of how many modals have been created so that the
+// container padding for the scrollbar is correctly set.
+let numberOfModalsCreated = 0;
```
Comment (Jyrno42) [inline src/browser.js]: Why are you tracking created modals? There is no limit on how many modals can be created in tg-modal. The current limitation is only for opened modals.

File: src/components/Modal.js [hunk 4]
```diff
@@ -5,6 +5,8 @@ import ModalHeader from './dom/ModalHeader';
 import ModalBody from './dom/ModalBody';
 import Backdrop from './dom/Backdrop';
 
+import toggleClass from '../toggle-class';
```
Comment (Jyrno42) [inline src/components/Modal.js]: remove: unused import

File: src/components/Modal.js [hunk 5]
```diff
@@ -40,10 +42,6 @@ class Modal extends Component {
         super(props);
 
         this.state = {};
-
-        if (props.isOpen) {
-            this.onToggle(props.isOpen, this.getToggleProps());
-        }
```
Comment (Jyrno42) [inline src/components/Modal.js]: Please explain the reasoning behind removing this.

File: src/components/Modal.js [hunk 6]
```diff
@@ -80,6 +78,11 @@ class Modal extends Component {
 
         if (this.props.isOpen && !this.props.isStatic) {
             if (this.props.onCancel) {
+
+                // Ensure that onToggle gets called.
+                this.onToggle(false, this.props);
+
+                // Call the user supplied function.
```
Comment (Jyrno42) [inline src/components/Modal.js]: Isn't onToggle called anyway, only after the next render though?

E.g.

1. User presses Escape
2. modal.onCancel gets called which triggers the `props.onCancel` hook
3. userland reacts to `props.onCancel` hook and sets modal open state to false
4. modal receives a props change where `isOpen` is now false
5. onToggle is called via `componentWillReceiveProps` 

Calling onToggle directly here removes the control from the userland, e.g. it would not be possible to `"ignore"` some onCancel requests anymore.

File: src/browser.js [hunk 7]
```diff
@@ -65,20 +69,34 @@ class BrowserModal extends Modal {
...
-        // Add body class
+        // Add body class and padding to scrollbar.
         if (typeof document !== 'undefined') {
             const container = document.body;
 
-            // Toggle open class
-            toggleClass(container, 'modal-open', state);
-
-            // Update body padding (for scrollbar)
+            // Add model when opening.
```
Comment (Jyrno42) [inline src/browser.js]: Change to something like: `increment numberOfModalsOpen when opening`

File: src/browser.js [hunk 8]
```diff
@@ -37,6 +40,7 @@ class BrowserModal extends Modal {
     }
 
     componentWillUnmount() {
+
```
Comment (Jyrno42) [inline src/browser.js]: why is this change necessary?

File: examples/components/examples/NestedLong.js [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                    <a className="btn btn-secondary" onClick={this.props.toggleCode}>Code</a>
+                </div>
+
+                {[...Array(this.state.numberOfModals).keys()].map(i => this.renderLongModal(i))}
+            </div>
+        );
+    }
+}
+
+export default LongNestedModalExample;
```
Comment (Jyrno42) [inline examples/components/examples/NestedLong.js]: missing newline at end of file

File: src/components/Modal.js (hunk 5 above)
Comment (schryer) [inline src/components/Modal.js]: Avoids double call.

File: src/components/Modal.js (hunk 6 above)
Comment (schryer) [inline src/components/Modal.js]: The problem turned out to be a bug in the parent class which was not calling the componentWillUnmount on the child class. This is fixed and the PR was updated.

File: src/browser.js (hunk 7 above)
Comment (schryer) [inline src/browser.js]: That change would tie the variable name into the comment which means that any future refactor will likely introduce a mystery comment bug that confuses the programmer more than informing them. The only purpose of this comment was to highlight that the adding and removing are performed in separated places within the same function.

Comment (Jyrno42) [inline src/browser.js]: then change `model` to `modal` :1234:
